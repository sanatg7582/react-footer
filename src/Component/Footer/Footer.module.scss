.FooterSection {
  width: 100%;
  background-color: #444444;
  float: left;
  color: #fff;
  height: 400px;
  padding-top: 50px;
  position: fixed;
  .FooterSectionChild {
    width: 23%;
    float: left;
    text-align: left;
    padding-left: 25px;
  }
}

.btn {
  background-color: #666666;
  color: #fff;
  border: none;
  padding: 8px 15px 8px 15px;
  border-radius: 5px;
  margin: 10px;
  font-weight: 700;
}
.footerShow {
  opacity: 1;
}
.footerHide {
  opacity: 0;
  transition: width 0.5s 0.5s, height 0.5s 0.5s, opacity 0.5s;
}
