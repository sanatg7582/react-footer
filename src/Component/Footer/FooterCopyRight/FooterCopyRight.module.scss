.footerCopyrRight {
  position: relative;
  color: #fff;
  float: left;
  top: 270px;
  span {
    padding: 0px 600px;
    font-size: 14px;
  }
  .tnccontent {
    padding: 25px;
    text-align: center;
  }
}
