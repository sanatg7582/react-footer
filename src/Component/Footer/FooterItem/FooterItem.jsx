import React from "react";
import PropTypes from "prop-types";
import styles from "./FooterItem.module.scss";
/**
 * FooterItem function for footer links
 * @param {string} FooterItems class
 * @param {string} title
 * return element
 */
const FooterItem = props => {
  return (
    <div className={styles.FooterItems}>
      <a href="/" title={props.title}>
        {props.title}
      </a>
    </div>
  );
};
// default props value
FooterItem.defaultProps = {
  title: ""
};
// propType validation
FooterItem.propTypes = {
  title: PropTypes.string
};
export default FooterItem;
