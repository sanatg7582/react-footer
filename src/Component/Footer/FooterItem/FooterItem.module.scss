.FooterItems {
  padding: 5px;
  a {
    color: #fff;
    font-size: 14px;
    text-decoration: none;
  }
  a:hover {
    text-decoration: underline;
  }
}
