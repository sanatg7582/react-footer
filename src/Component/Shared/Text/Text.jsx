import React from "react";
import PropTypes from "prop-types";
import styles from "./Text.module.scss";
/**
 * Text function for return text
 * @param {boolean} strong
 * @param {string} heading
 */
const Text = props => {
  const { strong, heading } = props;
  const headingClass = strong ? styles.footerHeading : "";
  return <span className={headingClass}>{heading}</span>;
};
Text.defaultProps = {
  strong: false
};
// PropsType validation here
Text.propTypes = {
  heading: PropTypes.string,
  strong: PropTypes.bool
};
export default Text;
