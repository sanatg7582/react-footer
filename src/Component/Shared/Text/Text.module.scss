.footerHeading {
  font-size: 17px;
  font-weight: 500;
  padding: 3px;
}
