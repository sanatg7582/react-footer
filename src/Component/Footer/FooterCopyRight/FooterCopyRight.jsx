import React from "react";
import PropTypes from "prop-types";
import styles from "./FooterCopyRight.module.scss";
import Text from "../../Shared/Text";
/**
 * FooterTermsAndCondition function for print label below copyright section
 * @param {string} tnccontent
 */
const FooterTermsAndCondition = tnccontent => {
  return (
    <div className={styles.tnccontent}>
      <Text heading={tnccontent} />
    </div>
  );
};
/**
 * This Fucntion for copyright content
 * @param {string} copyrighClass
 * @param {string} copyrightContent
 * return Element
 */
const FooterCopyRight = props => {
  const { copyrightcontent, tnccontent } = props;
  return (
    <div className={styles.footerCopyrRight}>
      <Text heading={copyrightcontent} />
      {FooterTermsAndCondition(tnccontent)}
    </div>
  );
};
// default props value
FooterCopyRight.defaultProps = {
  copyrightcontent: ""
};

// propType validation
FooterCopyRight.propTypes = {
  copyrightcontent: PropTypes.string
};
export default FooterCopyRight;
