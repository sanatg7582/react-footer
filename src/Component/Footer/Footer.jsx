import React, { Component } from "react";
import styles from "./Footer.module.scss";
import FooterItem from "./FooterItem";
import Text from "../Shared/Text";
import { footerData, footerSocialIcons } from "../../FooterData";
import FooterSocial from "./FooterSocial";
import FooterCopyRight from "./FooterCopyRight";

/**
 * Footer component render footer section with social logo and copyright text
 */
class Footer extends Component {
  constructor(props) {
    super(props);
    this.state = {
      toggle: true
    };
  }
  toggleFooter = () => {
    this.setState({
      toggle: !this.state.toggle
    });
  };
  render() {
    return (
      <>
        <button className={styles.btn} onClick={this.toggleFooter}>
          Toggle Footer
        </button>
        <div
          className={this.state.toggle ? styles.footerShow : styles.footerHide}
        >
          <div className={styles.FooterSection}>
            {footerData.map(data => {
              return (
                <div className={styles.FooterSectionChild} key={data.headingId}>
                  <Text heading={data.heading} strong={true} />
                  {data.Items.map(item => {
                    return <FooterItem title={item.title} key={item.ItemId} />;
                  })}
                </div>
              );
            })}
          </div>
          <FooterSocial footerSocialIcons={footerSocialIcons} />
          <div className="footercopyright">
            <FooterCopyRight
              copyrightcontent="Copyright &copy; 2001-2019 Coporate Express Canada. Inc.. a Staples company. All rights reserverd."
              tnccontent="Terms & Conditions    |   Privacy Policy    |   Staples Bussiness Advantage"
            />
          </div>
        </div>
      </>
    );
  }
}
export default Footer;
