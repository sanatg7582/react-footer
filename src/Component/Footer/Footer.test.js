import React from "react";
import { mount } from "enzyme";
import Footer from "./Footer";
import Text from "../Shared/Text";
import FooterItem from "./FooterItem";
import FooterCopyRight from "./FooterCopyRight";
describe("renders Footer without crashing ", () => {
  const wrapper = mount(<Footer />);

  it("renders without crashing", () => {
    expect(wrapper).toBeTruthy();
  });

  it("render footer section render should 5 times ", () => {
    expect(wrapper.find(".FooterSectionChild")).toHaveLength(5);
  });

  it("checking footer heading title, should be same", () => {
    expect(
      wrapper
        .find(Text)
        .first()
        .props().heading
    ).toEqual("Customer Service");
    expect(
      wrapper
        .find(Text)
        .at(1)
        .props().heading
    ).toEqual("Corporate Info");
    expect(
      wrapper
        .find(Text)
        .at(2)
        .props().heading
    ).toEqual("Staples Corporate Solution");
    expect(
      wrapper
        .find(Text)
        .at(3)
        .props().heading
    ).toEqual("New Customer");
    expect(
      wrapper
        .find(Text)
        .at(4)
        .props().heading
    ).toEqual("Join the Staples Team");

    // const headingtexts = wrapper.find(Text).map(node => node.text());
    // expect(headingtexts).toEqual([
    //   "Customer Service",
    //   "Corporate Info",
    //   "Staples Corporate Solution",
    //   "New Customer",
    //   "Join the Staples Team"
    // ]);
  });

  it("checking footer item lenght should be 18", () => {
    expect(wrapper.find(FooterItem)).toHaveLength(18);
  });

  it("checking FooterCopyRight prop value and length should be same", () => {
    expect(wrapper.find(FooterCopyRight).length).toEqual(1);
    expect(wrapper.find(FooterCopyRight).text("span")).toEqual(
      "Copyright Â© 2001-2019 Coporate Express Canada. Inc.. a Staples company. All rights reserverd.Terms & Conditions    |   Privacy Policy    |   Staples Bussiness Advantage"
    );
  });

  it("checking Footer toggle button click function", () => {
    expect(wrapper.find("button").simulate("click")).toHaveLength(1);
  });

  it("default footer section class", () => {
    wrapper.setState({ toggle: false });
    expect(
      wrapper
        .find("div")
        .first()
        .hasClass("footerHide")
    ).toEqual(true);
  });
  it("Snapshot test for Footer Component", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
