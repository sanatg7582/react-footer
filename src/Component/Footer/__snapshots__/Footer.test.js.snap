// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer without crashing  Snapshot test for Footer Component 1`] = `ReactWrapper {}`;
